sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -147,7 +154,15 @@ abstract class ExternalCatalog
    * Note: If the underlying implementation does not support altering a certain field,
    * this becomes a no-op.
    */
-  def alterTable(tableDefinition: CatalogTable): Unit
+  final def alterTable(tableDefinition: CatalogTable): Unit = {
```

Comment:
I'd suggest we leave it for now and watch other `alterTableXXX` events instead. I feel it's an overkill to have a heavy `alterTable` method to handling all table metadata updating, I think we will add more and more fine-grained alter table methods in the future, like `alterTableStats`, `alterTableDataSchema`, etc. and eventually this `alterTable` method will go away.

Reply:
This depends on which level of details we wanna collect in the event. Are there any guidelines or documentation of what events spark should monitor? 
Besides, partition changes are missing, I think it's necessary to also monitor these changes.

Reply:
@cloud-fan , since now we expose `alterTable` interface for other components to leverage, if we don't track this, then looks like we missed a piece of `ExternalCatalogEvent`s. I think for now we can add this `AlterTableEvent`, later on if we removed this method, then we can make this event a no-op (only kept for compatibility), what do you think?

@wzhfy , I was thinking to add partition related events, but I'm not clearly sure why this whole piece is missing and is it necessary to add partition related events? If we have an agreement on such events, I'm OK to add them.